### PR TITLE
[ros2]: Sync with master branch

### PR DESCRIPTION
**Pull request by JafarAbdi**
Please fast-forward merge using CLI

### CODE REVIEW

**Review by stephanie-eng — APPROVED** [review]
Thanks very much for helping with this!

File: core/test/test_move_to.cpp [hunk 1]
```diff
@@ -156,6 +157,34 @@ TEST_F(PandaMoveTo, poseIKFrameAttachedSubframeTarget) {
 	EXPECT_ONE_SOLUTION;
 }
 
+// This test require a running rosmaster
```
Comment (AndyZe) [inline core/test/test_move_to.cpp:160]: Doesn't apply to MoveIt2

```suggestion
```

File: core/test/test_move_to.cpp [hunk 2]
```diff
@@ -156,6 +157,34 @@ TEST_F(PandaMoveTo, poseIKFrameAttachedSubframeTarget) {
...
+		auto scene = std::make_shared<planning_scene::PlanningScene>(t.getRobotModel());
+		ref->setState(scene);
+
+		t.add(Stage::pointer(ref));
+		t.add(std::make_unique<ConnectMockup>());
+		t.add(std::make_unique<MonitoringGeneratorMockup>(ref));
+		return t;
+	};
+
+	// Segfaults when introspection is enabled
```
Comment (AndyZe) [inline core/test/test_move_to.cpp:176]:
```suggestion
	// Does not segfault when introspection is enabled
```

File: core/src/solvers/cartesian_path.cpp [hunk 3]
```diff
@@ -41,6 +41,7 @@
 
 #include <moveit/planning_scene/planning_scene.h>
 #include <moveit/trajectory_processing/time_parameterization.h>
+#include <moveit/kinematics_base/kinematics_base.h>
 #include <moveit/robot_state/cartesian_interpolator.h>
```
Comment (AndyZe) [inline core/src/solvers/cartesian_path.cpp:45]:
```suggestion
#include <moveit/kinematics_base/kinematics_base.h>
#include <moveit/planning_scene/planning_scene.h>
#include <moveit/robot_state/cartesian_interpolator.h>
#include <moveit/trajectory_processing/time_parameterization.h>
```

File: core/test/test_move_to.cpp (hunk 2 above)
Comment (JafarAbdi) [reply core/test/test_move_to.cpp:176]: Yeah, I'll update the test description to mention that issue it's fixing

**Review by rhaschke — COMMENTED** [review]
Contentwise fine. But, please perform merge commits instead of cherry picks to more easily see what was already merged. We did that in the past in this repo as well.
See here: https://github.com/ros-planning/moveit_task_constructor/pull/373

**Pull request closed without merging**